main.tf

```diff
@@ -1,13 +1,28 @@
-resource "random_integer" "example" {
+# -----------------------------------------------------------------------------------------------------------------------
+# Subscribe the Acccount to Security Hub
+# -----------------------------------------------------------------------------------------------------------------------
+resource "aws_securityhub_account" "enable_us_east_1" {
   count = module.this.enabled ? 1 : 0
+}
+
+# -----------------------------------------------------------------------------------------------------------------------
+# Optionally subscribe to Security Hub Standards 
+# https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-standards.html
+# -----------------------------------------------------------------------------------------------------------------------
+resource "aws_securityhub_standards_subscription" "enable_cis_1_2_0" {
```

Comment:
I'm leaning towards using a variable called `rulesets` which is an empty list by default. 

The user could pass the rulesets and based on the provider the region and partition would be used to compute the ARN. 

e.g. 
```hcl
module "security_hub_us_east_1" {
  rulesets = [ "ruleset/cis-aws-foundations-benchmark/v/1.2.0" ]
  providers = {
    aws = aws.us-east-1
  }
}
```

And the arn would be:

```hcl
data "aws_region" "this" {}
data "aws_partition" "this" {}
locals {
for l in local.label_order : local.id_context[l] if length(local.id_context[l]) > 0
  ruleset_arns = [for ruleset in var.rulesets : "arn:${data.aws_partition.this.partition}:securityhub:${data.aws_region.this.name}::${ruleset}"]
}

resource "aws_securityhub_standards_subscription" "standard" {
   for_each = local.ruleset_arns
   standards_arn = each.key
 }	

```

Reply:
When we call the module, our stack configuration would look like:

We define a stack per region:
```
projects:
  globals:
    stage: dev
    env: ue2
    region: us-east-2

  terraform:
    security_hub:
      vars:
         rulesets:
         - "ruleset/cis-aws-foundations-benchmark/v/1.2.0"